decompress_zip.go

```diff
@@ -46,7 +46,7 @@ func (d *ZipDecompressor) Decompress(dst, src string, dir bool) error {
 		}
 
 		if f.FileInfo().IsDir() {
-			if dir {
+			if !dir {
```

Comment:
I think this is how we were seeing this error [here](https://github.com/hashicorp/nomad/issues/923)